packages/lib/src/vis/line/DataCurve.tsx

```diff
@@ -52,15 +78,34 @@ function DataCurve(props: Props) {
     onDataPointLeave,
   } = props;
 
-  const [dataGeometry] = useState(() => new BufferGeometry());
-  const points = useCanvasPoints(abscissas, ordinates, errors);
+  const showLine = visible && curveType !== CurveType.GlyphsOnly;
+  const showGlyphs = visible && curveType !== CurveType.LineOnly;
+
+  const size = useThree((state) => state.size);
   const invalidate = useThree((state) => state.invalidate);
+  const points = useCanvasPoints(abscissas, ordinates, errors);
+  const bufferGeometry = useState(() => new BufferGeometry())[0];
+  const lineGeometry = useState(() => new LineGeometry())[0];
 
+  function getFlatData(): number[] {
+    return lineWidth === 1 ? [] : flattenCoordinates(points.data);
+  }
+  const flatData = useMemo(getFlatData, [lineWidth, points.data]);
+  if (lineWidth === 1 || showGlyphs) {
+    bufferGeometry.setFromPoints(points.data);
+  } else {
+    lineGeometry.setPositions(flatData);
```

Comment:
There's a bit of a performance issue here: [one loop](https://github.com/silx-kit/h5web/pull/1347/files#diff-87080b3b5aeb767584fe4e9df8460c5f30453d44d12ea990e960c03c93326a6bR54) for to go from `[vec1, vec2, ...]` to `[x1, y1, z1, x2, y2, z2, ...]`, then [another loop](https://github.com/pmndrs/three-stdlib/blob/main/src/lines/LineGeometry.js#L18) to go from that to line segments.

Any chance we could initialise a buffer geometry manually, in order to convert the input vectors array to a buffer of line segments in a single pass?

Reply:
I think this logic is a little buggy: we need both geometries set when `showGlyphs` and `lineWith != 1`. I could use  `LineSegments2` and conditionally compute two arrays with one loop.